tests/test_unicorn.py

```diff
@@ -1306,6 +1313,15 @@ def test_umull_z(self):
         self.assertEqual(self.rf.read('R2'), (mul >> 32) & Mask(32))
         self._checkFlagsNZCV(0, 1, pre_c, pre_v)
 
+    def test_thumb_mode_emulation(self):
+        asm = "add r0, r1, r2"
+        self._setupCpu(asm, mode=CS_MODE_THUMB)
+        self.rf.write('R0', 0)
+        self.rf.write('R1', 0x1234)
+        self.rf.write('R2', 0x5678)
+        emulate_next(self.cpu)
+        self.assertEqual(self.rf.read('R0'), 0x1234 + 0x5678)
```

Comment:
Hmm can we also add an assertion for cpu mode here?

Reply:
I replaced your check on the Manticore cpu by a check on the Unicorn cpu mode. :smiley: